### PR TITLE
Remove damage reductions from Cassidy's roll and deadeye V2

### LINKED ISSUE #104: Remove damage reduction during roll and deadeye from Cassidy
OW2 introduced damage reduction when Cassidy is rolling or using deadeye. OW1 didn't have any.

### PR DESCRIPTION
**Pull request by MrKingMichael**
Fixes #104, hope I did it right this time.

### CODE REVIEW

File: src/constants/ow2_constants.opy [hunk 1]
```diff
@@ -79,4 +79,8 @@
 #!define OW2_WINSTON_ULT_COST 1850
 
 # Wrecking ball
-#!define OW2_WRECKING_BALL_PILEDRIVER_CD 8
\ No newline at end of file
+#!define OW2_WRECKING_BALL_PILEDRIVER_CD 8
+
+# Cassidy
+#!define OW2_CASSIDY_ROLL_DAMAGE_REDUCTION 50
+#!define OW2_CASSIDY_DEADEYE_DAMAGE_REDUCTION 40
```
Comment (MaxwellJung) [inline src/constants/ow2_constants.opy:86]: Rename Cassidy to Mccree becasue OverPy uses Mccree to refer to Cassidy.

Also move this section of code to line ~66 (after Bastion constants) to keep the hero constants alphabetical.

File: src/ow1/heroes/mccree.opy [hunk 2]
```diff
@@ -42,3 +42,21 @@ rule "[mccree.opy]: Stun flashed enemy":
     @Condition attacker.getCurrentHero() == Hero.MCCREE or attacker.getCurrentHero() == Hero.GENJI
 
     victim.setStatusEffect(eventPlayer, Status.STUNNED, OW1_MCCREE_FLASHBANG_STUN_DURATION)
+
+rule "[mccree.opy]: Remove damage reduction during roll":
+    @Event eachPlayer
+    @Hero mccree
+    @Condition eventPlayer.isUsingAbility1() == true
+
+    eventPlayer.setDamageReceived(100/(100-OW2_CASSIDY_ROLL_DAMAGE_REDUCTION)*100)
```
Comment (MaxwellJung) [inline src/ow1/heroes/mccree.opy:51]:
```suggestion
    eventPlayer.setDamageReceived(percent(100/(100-OW2_CASSIDY_ROLL_DAMAGE_REDUCTION)))
```

the percent() function is defined in utilities.opy so we don't have to write *100 in the source code.

File: src/ow1/heroes/mccree.opy [hunk 3]
```diff
@@ -42,3 +42,21 @@ rule "[mccree.opy]: Stun flashed enemy":
...
+    waitUntil(not eventPlayer.isUsingAbility1(), 9999)
+    eventPlayer.setDamageReceived(100)
+
+rule "[mccree.opy]: Remove damage reduction during deadeye":
+    @Event eachPlayer
+    @Hero mccree
+    @Condition eventPlayer.isUsingUltimate() == true
+
+    eventPlayer.setDamageReceived(100/(100-OW2_CASSIDY_DEADEYE_DAMAGE_REDUCTION)*100)
```
Comment (MaxwellJung) [inline src/ow1/heroes/mccree.opy:60]: same suggestion as above

**Review by MaxwellJung — CHANGES_REQUESTED** [review]

**Comment by MaxwellJung** [comment]
Other than those changes, code looks good.

After confirming it works, I'll merge this to main.

File: src/constants/ow2_constants.opy [hunk 4]
```diff
@@ -79,4 +79,8 @@
 #!define OW2_WINSTON_ULT_COST 1850
 
 # Wrecking ball
-#!define OW2_WRECKING_BALL_PILEDRIVER_CD 8
\ No newline at end of file
+#!define OW2_WRECKING_BALL_PILEDRIVER_CD 8
```
[the remark this answers is not in the document]
Comment (MrKingMichael) [reply src/constants/ow2_constants.opy:86]: Not sure, might be a bug with GitHub on browser.

**Comment by MrKingMichael** [comment]
Done, please let me know if it needs anything else.

**Review by MaxwellJung — APPROVED** [review]
Looks good

**Pull request merged**